src/main/java/gift/dto/ProductRequest.java

```diff
@@ -1,11 +1,25 @@
 package gift.dto;
 
+import gift.validation.NoKakao;
+import jakarta.validation.constraints.*;
+
 public class ProductRequest {
     private Long id;
+
+    @NotBlank(message = "상품 이름은 필수입니다.")
+    @Size(max = 15, message = "상품 이름은 최대 15자까지 입력 가능합니다.")
+    @Pattern(
+            regexp = "^[a-zA-Z0-9가-힣 ()\\[\\]+\\-&/_]*$",
+            message = "상품 이름에는 허용되지 않는 특수 문자가 포함되어 있습니다."
+    )
+    @NoKakao
     private String name;
```

Comment:
정확한 구현과 명확한 메시지 좋아요 👍 
이건 팁인데 저는 `^[a-zA-Z0-9가-힣 ()\\[\\]+\\-&/_]*$` 이런식으로 한 눈에 알아보기 힘든 부분들이 있을 때 이해를 돕기위한 주석을 남기는 편이에요.
```suggestion
    @NotBlank(message = "상품 이름은 필수입니다.")
    @Size(max = 15, message = "상품 이름은 최대 15자까지 입력 가능합니다.")
    @Pattern(
            regexp = "^[a-zA-Z0-9가-힣 ()\\[\\]+\\-&/_]*$", // 허용되는 특수 문자 (, ), [, ], +, -, &, /, _
            message = "상품 이름에는 허용되지 않는 특수 문자가 포함되어 있습니다."
    )
    @NoKakao
    private String name;
```